### DISCUSSION TITLE
There are any lit library module from cdn?

### DISCUSSION THREAD

**Discussion by dbwodlf3** [lit/lit#3462] [Q&A]
I would like to sharing lit library codes on some pages.

like below.

```js
<script src="..../lit.min.js"></script>
<script>
    console.log(lit);
</script>
```

**Comment by bicknellr** [comment]
One of the ways Lit is distributed is as bundled modules, you can find more info at <https://lit.dev/docs/getting-started/#use-bundles>. Though, if you're building your site using npm for client-side dependencies, we recommend you stick with the npm packages.

**Reply by dbwodlf3** [reply]
Thanks, but this is only run ```<script type="module"></script>```. and doesn't expose lit module as global name..

**Reply by miltonhowe** [reply]
Is there any detailed documentation on how to set up the imports using the bundles served by jsdelivr? 
I am not able to get it to work despite a lot of trying. 
I was previously able to use unpkg by including a long list of import mappings in a `<script type="importmap">` tag.
I am getting errors trying to load from those bundles, in particular with lit/decorators.js.
The following first line works, subsequent lines fail
```
import {LitElement, html, css, nothing} from 'lit';
import {customElement, property} from 'lit/decorators.js';
import {until} from 'lit/directives/until.js';
import {when} from 'lit/directives/when.js';
```
when i use the following importmap settings: (adding maps to deeper in the tree have also all failed)
```
<script type="importmap">
{
  "imports": {
    "lit": "https://cdn.jsdelivr.net/gh/lit/dist@2.6.1/all/lit-all.min.js"
  }
}
</script>
```

if I try to import everything directly from `https://cdn.jsdelivr.net/gh/lit/dist@2.6.1/all/lit-all.min.js`, I get this message
`The requested module 'https://cdn.jsdelivr.net/gh/lit/dist@2.6.1/all/lit-all.min.js' does not provide an export named 'customElement'`

**Reply by p-ob** [reply]
> Thanks, but this is only run ```<script type="module"></script>```. and doesn't expose lit module as global name..

You could probably do this yourself. Just would want something to observe that indicate lit has fully loaded and instantiated on the global object (like a promise). 

The modular approach of lit was always to reduce the amount of bloat, since tree-shaking isn't a thing in the browser like build tooling. The "kitchen-sink" approach is nice for quick hacking, but with the detriment of loading things you don't need.

**Reply by p-ob** [reply]
> Is there any detailed documentation on how to set up the imports using the bundles served by jsdelivr? 
> I am not able to get it to work despite a lot of trying. 
> I was previously able to use unpkg by including a long list of import mappings in a `<script type="importmap">` tag.
> I am getting errors trying to load from those bundles, in particular with lit/decorators.js.
> The following first line works, subsequent lines fail
> ```
> import {LitElement, html, css, nothing} from 'lit';
> import {customElement, property} from 'lit/decorators.js';
> import {until} from 'lit/directives/until.js';
> import {when} from 'lit/directives/when.js';
> ```
> when i use the following importmap settings: (adding maps to deeper in the tree have also all failed)
> ```
> <script type="importmap">
> {
>   "imports": {
>     "lit": "https://cdn.jsdelivr.net/gh/lit/dist@2.6.1/all/lit-all.min.js"
>   }
> }
> </script>
> ```
> 
> if I try to import everything directly from `https://cdn.jsdelivr.net/gh/lit/dist@2.6.1/all/lit-all.min.js`, I get this message
> `The requested module 'https://cdn.jsdelivr.net/gh/lit/dist@2.6.1/all/lit-all.min.js' does not provide an export named 'customElement'`

You created an importmap entry for the default module (`import * from "lit";`) but you didn't tell the browser to handle pathed imports beneath. You'll need an entry for `"lit/"` in your importmap I think.

**Edit:** actually, the "all" bundle will export everything. So all imports would be from "lit", not "lit/directives/until.js". I don't believe the decorators are exported in this bundle though. You'd have to use the alternatives.

**Reply by augustjk** [reply]
the edit above is correct. the jsdelivr bundles are primarily meant for prototyping use in tool-less environments. Decorators are not a native JS feature (yet) so are not included in bundles.
